src/de.f90

```diff
@@ -23,18 +23,20 @@ module de
 
 
   !Main differential evolution routine.
-  subroutine diver(func, lowerbounds, upperbounds, path, nDerived, discrete, partitionDiscrete,            &
-                   maxciv, maxgen, NP, F, Cr, lambda, current, expon, bndry, jDE, lambdajDE,               &
-                   convthresh, convsteps, removeDuplicates, doBayesian, prior, maxNodePop, Ztolerance,     &
-                   savecount, resume, outputSamples, init_population_strategy, discard_unfit_points,       &
-                   max_initialisation_attempts, max_acceptable_value, seed, context, verbose)
+  function diver(func, lowerbounds, upperbounds, path, nDerived, paramsPlus, discrete, partitionDiscrete, &
+                 maxciv, maxgen, NP, F, Cr, lambda, current, expon, bndry, jDE, lambdajDE,                &
+                 convthresh, convsteps, removeDuplicates, doBayesian, prior, maxNodePop, Ztolerance,      &
+                 savecount, resume, outputSamples, init_population_strategy, discard_unfit_points,        &
+                 max_initialisation_attempts, max_acceptable_value, seed, context, verbose)
 
     use iso_c_binding, only: c_ptr
 
+    real(dp)                         :: diver                   !result (minimum value obtained from func)
     procedure(MinusLogLikeFunc)      :: func                    !the likelihood function to be minimised -- assumed to be -ln(likelihood)
     real(dp), dimension(:), intent(in) :: lowerbounds, upperbounds !boundaries of parameter space
     character(len=*), intent(in)     :: path                    !path to save samples, resume files, etc
     integer, intent(in), optional    :: nDerived                !input number of derived quantities to output
+    real(dp), intent(out), dimension(:), optional :: paramsPlus !values of parameters and derived quantities at mimimum
```

Comment:
> So the big change here is that diver is now a function where the output is the final best fit likelihood and paramsPlus contains all the other parameter info for the final best fit vector? 

Correct

>  I think this is a reasonable choice, but since that no longer matches the docuementation in the paper, we should probably document this, either in the readme or maybe set up a wiki? I can work on that if you agree that it's a good idea.

That would be awesome, thanks :smile: Either in a wiki linked from the readme or in the readme directly would be great.  It's probably best to just seed it with the existing documentation from the paper, and then update that.

> I also think that the name is a bit vague. Since it's intent(out) how about something like finalParams or bestFitParams? (Or best_fit_params; we seem to have some variability in our variable names)

Yeah the name is crap.  I had trouble coming up with something sensible that wasn't a 100 characters long.  I think what you suggest is good if combined with this suggestion:

>  I don't mind the derived parameters being added on so much since they are parameters of a sort, but it would also be reasonable to give them their own output array.

which would make things a lot clearer and simpler.  I'll implement this.

Reply:
OK, done

Reply:
Yes, this is a lot clearer, thank you! 
I'll get started on the wiki.